spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsBinderSupportAutoConfiguration.java

```diff
@@ -60,29 +64,61 @@ public KafkaStreamsBinderConfigurationProperties binderConfigurationProperties(K
 	}
 
 	@Bean("streamConfigGlobalProperties")
-	public Map<String, Object> streamConfigGlobalProperties(KafkaStreamsBinderConfigurationProperties binderConfigurationProperties) {
-		Map<String, Object> props = new HashMap<>();
-		props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, binderConfigurationProperties.getKafkaConnectionString());
-		props.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.ByteArraySerde.class.getName());
-		props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.ByteArraySerde.class.getName());
-		props.put(StreamsConfig.APPLICATION_ID_CONFIG, binderConfigurationProperties.getApplicationId());
+	public Map<String, Object> streamConfigGlobalProperties(KafkaStreamsBinderConfigurationProperties binderConfigurationProperties,
+															Environment environment) {
+
+		//Initialize configuration with default boot auto configuration properties
+		Map<String, Object> streamsConfiguration =
+				new HashMap<>(binderConfigurationProperties.getKafkaProperties().buildStreamsProperties());
+
+		// Override Spring Boot bootstrap server setting if left to default with the value
+		// configured in the binder
+		if (ObjectUtils.isEmpty(streamsConfiguration.get(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG))) {
+			streamsConfiguration.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, binderConfigurationProperties.getKafkaConnectionString());
+		}
+		else {
+			Object boostrapServersConfig = streamsConfiguration.get(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG);
+			if (boostrapServersConfig instanceof List) {
+				@SuppressWarnings("unchecked")
+				List<String> bootStrapServers = (List<String>) streamsConfiguration
+						.get(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG);
+				if (bootStrapServers.size() == 1 && bootStrapServers.get(0).equals("localhost:9092")) {
+					streamsConfiguration.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, binderConfigurationProperties.getKafkaConnectionString());
+				}
+			}
+		}
+
+		String binderProvidedApplicationId = binderConfigurationProperties.getApplicationId();
+		if (StringUtils.isEmpty(streamsConfiguration.get(StreamsConfig.APPLICATION_ID_CONFIG))) {
+			streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, binderProvidedApplicationId);
+		}
+		else {
+			String applicationId = (String)streamsConfiguration.get(StreamsConfig.APPLICATION_ID_CONFIG);
+			if (applicationId.equals(environment.getProperty("spring.application.name")) &&
+					!StringUtils.isEmpty(binderProvidedApplicationId)) {
+				streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, binderProvidedApplicationId);
+			}
+		}
```

Comment:
Shouldn't a binder-specific `applicationId` ALWAYS take precedence over the boot one, regardless of boot's value?